parsers/osx_file_parser.py

```diff
@@ -53,3 +54,115 @@ def Parse(self, cmd, args, stdout, stderr, return_val, time_taken,
     serial_number = hardware_list["serial_number"]
 
     yield rdfvalue.HardwareInfo(serial_number=serial_number)
+
+class OSXLaunchdPlistParser(parsers.FileParser):
+  """Parse Launchd plist files into LaunchdPlist objects."""
+
+  output_types = ["LaunchdPlist"]
+  supported_artifacts = ["OSXLaunchAgents", "OSXLaunchDaemons"]
+
+  def Parse(self, stat, file_object, knowledge_base):
+    """Parse the Plist file."""
+    _ = knowledge_base
+    kwargs = {}
+    kwargs["aff4path"] = stat.aff4path
+
+    direct_copy_items = ["Label", "Disabled", "UserName", "GroupName",
+                         "Program", "StandardInPath", "StandardOutPath",
+                         "StandardErrorPath", "LimitLoadToSessionType",
+                         "EnableGlobbing", "EnableTransactions", "OnDemand",
+                         "RunAtLoad", "RootDirectory", "WorkingDirectory",
+                         "Umask", "TimeOut", "ExitTimeOut", "ThrottleInterval",
+                         "InitGroups", "StartOnMount", "StartInterval",
+                         "Debug", "WaitForDebugger", "Nice", "ProcessType",
+                         "AbandonProcessGroup", "LowPriorityIO",
+                         "LaunchOnlyOnce"]
+
+    string_array_items = ["LimitLoadToHosts", "LimitLoadFromHosts",
+                          "LimitLoadToSessionType", "ProgramArguments",
+                          "WatchPaths", "QueueDirectories"]
+
+    flag_only_items = ["SoftResourceLimits", "HardResourceLimits", "Sockets"]
+
+    plist = {}
+
+    try:
+      plist = binplist.readPlist(file_object)
+    except binplist.FormatError:
+      plist["Label"] = "Could not parse plist."
+
+    # These are items that can be directly copied
+    for key in direct_copy_items:
+      kwargs[key] = plist.get(key)
+
+    # These are arrays of strings that need to be combined
+    for key in string_array_items:
```

Comment:
The string array type should be a repeated string in the protobuf definition. Then you can just directly assign the list to the rdfvalue:

if isinstance(elements, list):
  kwargs[key] = elements
# Can it actually happen that this is not a list?
else:
  kwargs[key] = [elements]